vochain/transaction.go

```diff
@@ -560,19 +560,25 @@ func AdminTxCheck(vtx *models.Tx, txBytes, signature []byte, state *State) error
 	if signature == nil || tx == nil || txBytes == nil {
 		return fmt.Errorf("missing signature and/or admin transaction")
 	}
-	// get oracles
-	oracles, err := state.Oracles(false)
-	if err != nil || len(oracles) == 0 {
-		return fmt.Errorf("cannot check authorization against a nil or empty oracle list")
+	// get treasurer
+	treasurer, err := state.Treasurer(false)
+	if err != nil {
+		return fmt.Errorf("cannot check authorization")
 	}
 
-	if authorized, addr, err := verifySignatureAgainstOracles(
-		oracles, txBytes, signature); err != nil {
-		return err
-	} else if !authorized {
-		return fmt.Errorf("unauthorized to perform an adminTx, address: %s", addr.Hex())
+	pubKey, err := ethereum.PubKeyFromSignature(txBytes, signature)
+	if err != nil {
+		return fmt.Errorf("cannot extract public key from signature: %w", err)
+	}
+	addr, err := ethereum.AddrFromPublicKey(pubKey)
+	if err != nil {
+		return fmt.Errorf("cannot extract address from public key: %w", err)
+	}
+	log.Debugf("checking admin signed tx %+v by addr %s", tx, addr.String())
+	log.Debugf("got treasurer addr %s", common.BytesToAddress(treasurer.Address).String())
+	if !bytes.Equal(addr.Bytes(), treasurer.Address) {
+		return fmt.Errorf("signature extracted address does not match with treasurer address")
```

Comment:
Since this is an error we are returning to the users I'd do something more high level such as `not authorized for executing admin transactions`